src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -235,8 +237,17 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 
 		for (attributeIndex = 0; attributeIndex < attributeCount; attributeIndex++)
 		{
-			AttrNumber attributeNumber = attributeNumberArray[attributeIndex];
-			if (distributionColumn->varattno == attributeNumber)
+			if (distributionColumn->varattno != attributeNumberArray[attributeIndex])
```

Comment:
I think having an explicit variable assignement is more convinient wrt Citus conventions. In this case, it'd be better if we keep `AttrNumber attributeNumber` and continue to use it.